FirebaseAI/Sources/GenerateContentResponse.swift

```diff
@@ -299,6 +302,123 @@ public struct PromptFeedback: Sendable {
+  /// Google Search*.
+  public let groundingChunks: [GroundingChunk]
+  /// A list of ``GroundingSupport`` structs. Each object details how specific segments of the
+  /// model's response are supported by the `groundingChunks`.
+  ///
+  /// > Important: If using Grounding with Google Search, you are required to comply with the
+  /// [Service Specific Terms](https://cloud.google.com/terms/service-terms) for *Grounding with
+  /// Google Search*.
+  public let groundingSupports: [GroundingSupport]
+  /// Google search entry point for web searches.
+  /// This contains an HTML/CSS snippet that **must** be embedded in an app to display a Google
+  /// Search Entry point for follow-up web searches related to the model's "Grounded Response".
+  ///
+  /// > Important: If using Grounding with Google Search, you are required to comply with the
+  /// [Service Specific Terms](https://cloud.google.com/terms/service-terms) for *Grounding with
+  /// Google Search*.
+  public let searchEntryPoint: SearchEntryPoint?
+
+  /// A struct representing the Google Search entry point.
+  ///
+  /// > Important: If using Grounding with Google Search, you are required to comply with the
+  /// [Service Specific Terms](https://cloud.google.com/terms/service-terms) for *Grounding with
+  /// Google Search*.
+  @available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+  public struct SearchEntryPoint: Sendable {
+    /// An HTML/CSS snippet that can be embedded in your app. The snippet is designed to avoid
+    /// undesired interaction with the rest of the page's CSS.
+    ///
+    /// To ensure proper rendering, it's recommended to display this content within a `WKWebView`.
+    public let renderedContent: String?
+  }
+
+  /// Represents a chunk of retrieved data that supports a claim in the model's response. This is
+  /// part
+  /// of the grounding information provided when grounding is enabled.
+  ///
+  /// > Important: If using Grounding with Google Search, you are required to comply with the
+  /// [Service Specific Terms](https://cloud.google.com/terms/service-terms) for *Grounding with
+  /// Google Search*.
+  @available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+  public struct GroundingChunk: Sendable {
+    /// Contains details if the grounding chunk is from a web source.
+    public let web: WebGroundingChunk?
+  }
+
+  /// A grounding chunk sourced from the web.
+  ///
+  /// > Important: If using Grounding with Google Search, you are required to comply with the
+  /// [Service Specific Terms](https://cloud.google.com/terms/service-terms) for *Grounding with
+  /// Google Search*.
+  @available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+  public struct WebGroundingChunk: Sendable {
+    /// The URI of the retrieved web page.
+    public let uri: String?
+    /// The title of the retrieved web page.
+    public let title: String?
```

Comment:
> Do all of the new optional fields need to be optional?
-- @paulb777 

@dlarocque Have you seen any cases where `uri` or `title` are nil? I don't think a developer could reasonably use the `WebGroundingChunk` if both of these are nil.

Reply:
I have seen cases where `title` is nil. I have never seen a case where `uri` is nil. We could make `uri` required but I lean towards keeping these optional since the fields in the proto are explicitly `optional`.

If both `uri` and `title` are nil, the response won't have a `"web"` object.

Reply:
SGTM to keep them optional. Thanks for investigating.